### PR TITLE
import into: impl import from server disk

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #42930 

Problem Summary:

### What is changed and how it works?
- change local storage to support non-recusive walk
- add import from server disk, which requires SEM disabled and FILE privilege
- add check about options, some options are only allowed for CSV format
- remove useless codes

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - run 2 tidb on 2 different machine
  - connect to the tidb which is not the owner, and create some csv on that tidb, then import from server disk
  - import success and result is correct
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/parse_test.go [hunk 1]
```diff
@@ -260,3 +260,33 @@ func TestParseRawURL(t *testing.T) {
...
+		args   args
+		want   bool
+		errStr string
+	}{
+		{"local", args{":"}, false, "missing protocol scheme"},
+		{"local", args{"~/tmp/file"}, true, ""},
+		{"local", args{"/tmp/file"}, true, ""},
+		{"local", args{"local:///tmp/file"}, true, ""},
+		{"local", args{"file:///tmp/file"}, true, ""},
+		{"local", args{"s3://bucket/tmp/file"}, false, ""},
```
Comment (GMHDBJD) [inline br/pkg/storage/parse_test.go:279]: test current dir: "."

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: disttask/loaddata/job.go [hunk 2]
```diff
@@ -65,6 +68,17 @@ func NewDistImporterCurrNode(param *importer.JobImportParam, plan *importer.Plan
 	}, nil
 }
 
+// NewDistImporterServerFile creates a new DistImporter to import given files on current node.
+// we also run import on current node.
+func NewDistImporterServerFile(param *importer.JobImportParam, plan *importer.Plan, stmt string, ecp map[int32]*checkpoints.EngineCheckpoint) (*DistImporter, error) {
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply disttask/loaddata/job.go:73]: left a todo, will change it later

File: br/pkg/storage/parse_test.go (hunk 1 above)
Comment (D3Hunter) [reply br/pkg/storage/parse_test.go:282]: https://github.com/pingcap/tidb/pull/44273/commits/02d514e52258e42698a3cfabd514b53f3fa0edeb

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — COMMENTED** [review]
16 / 30 files viewed

File: executor/importer/import.go [hunk 3]
```diff
@@ -756,17 +777,41 @@ func (e *LoadDataController) InitDataFiles(ctx context.Context) error {
...
+		if !e.InImportInto {
+			return exeerrors.ErrLoadDataFromServerDisk.GenWithStackByArgs(e.Path)
+		}
+
+		if !filepath.IsAbs(e.Path) {
+			return exeerrors.ErrLoadDataInvalidURI.GenWithStackByArgs("file location should be absolute path when import from server disk")
+		}
+		// we add this check for security, we don't want user import any sensitive system files,
+		// most of which is readable text file and don't have a suffix, such as /etc/passwd
+		if !slices.Contains([]string{".csv", ".sql", ".parquet"}, strings.ToLower(filepath.Ext(e.Path))) {
```
Comment (lance6716) [inline executor/importer/import.go:793]: doc should notify the user that we only support these suffixes

**Review by lance6716 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**